### PR TITLE
TestRunner unified

### PR DESCRIPTION
**Pull request by arkadius**

### CODE REVIEW

File: docs/developers_guide/Testing.md [hunk 1]
```diff
@@ -35,27 +35,42 @@ val scenario =
...
 ```
 
-:::caution
-For flink runner test must extend FlinkSpec for now.
-:::
+- liteBased - based on Lite engine, no other setup needed, provides interface to communicate with engine using raw classes (similar interface as using flinkBased):
+```scala
+import pl.touk.nussknacker.engine.lite.util.test.LiteKafkaTestScenarioRunner._
+val testScenarioRunner = TestScenarioRunner
+  .kafkaLiteBased()
```
Comment (mproch) [inline docs/developers_guide/Testing.md:54]: liteBased here?

File: docs/developers_guide/Testing.md [hunk 2]
```diff
@@ -35,27 +35,42 @@ val scenario =
...
+import pl.touk.nussknacker.engine.flink.util.test.FlinkTestScenarioRunner._
+val testScenarioRunner = TestScenarioRunner
+  .flinkBased(config, flinkMiniCluster)
+  .build()
 ```
 
-:::caution
-For flink runner test must extend FlinkSpec for now.
-:::
+- liteBased - based on Lite engine, no other setup needed, provides interface to communicate with engine using raw classes (similar interface as using flinkBased):
```
Comment (mproch) [inline docs/developers_guide/Testing.md:50]:
```suggestion
- liteBased - based on Lite engine, no other setup needed, provides interface to communicate with engine using raw classes (not using any Kafka API, similar interface as using flinkBased):
```

File: docs/developers_guide/Testing.md [hunk 3]
```diff
@@ -35,27 +35,42 @@ val scenario =
 ```
 
 ### Creating test scenario runner
-Scenario should be executed inside a runner. `NuTestScenarioRunner` gives you another DLS for building runners.
+
+Scenario should be executed inside a runner. `TestScenarioRunner` gives you another DLS for building runners.
```
Comment (mproch) [inline docs/developers_guide/Testing.md:39]: We can add mention that sample components names are provided in `TestScenarioRunner` object

**Review by mproch — APPROVED** [review]
Cool, added just minor comments

File: docs/developers_guide/Testing.md (hunk 1 above)
Comment (arkadius) [reply docs/developers_guide/Testing.md:54]: :+1:

File: docs/developers_guide/Testing.md (hunk 2 above)
Comment (arkadius) [reply docs/developers_guide/Testing.md]: I changed. The order of test runners descriptions were changed after this.

File: docs/developers_guide/Testing.md (hunk 3 above)
Comment (arkadius) [reply docs/developers_guide/Testing.md:39]: I extended `Running scenario with data` section - I mentioned there about test sources and sinks. This one sentence doesn't look good:
> Results are collected using `sink` component in `liteBased` case and `invocationCollector` in `flinkBased` case
but it's showing current situation :/

**Pull request merged**